autorest/codegen/templates/setup.py.jinja2

```diff
@@ -1,11 +1,75 @@
-{% set name = code_model.options["package_name"] or code_model.class_name %}
-{% set azure_mgmt_core_import = ', "azure-mgmt-core<2.0.0,>=1.2.1"' if code_model.options["azure_arm"] else "" %}
 # coding=utf-8
-{{ code_model.options['license_header'] }}
+{{ license_header }}
 # coding: utf-8
-
+{% if package_mode %}
+import os
+import re
+{% endif -%}
 from setuptools import setup, find_packages
 
+{% if package_mode %}
+# Change the PACKAGE_NAME only to change folder and different name
+PACKAGE_NAME = "{{ package_name }}"
+PACKAGE_PPRINT_NAME = "{{ package_pprint_name }}"
+
+# a-b-c => a/b/c
+package_folder_path = PACKAGE_NAME.replace("-", "/")
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(package_folder_path, "_version.py"), "r") as fd:
+    version = re.search(
+        r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE
+    ).group(1)
+
+if not version:
+    raise RuntimeError("Cannot find version information")
+
+setup(
+    name=PACKAGE_NAME,
+    version=version,
+    description="Microsoft {{ package_pprint_name }} Client Library for Python",
+    long_description=open("README.md", "r").read(),
+    long_description_content_type="text/markdown",
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="azpysdkhelp@microsoft.com",
+    url="https://github.com/Azure/azure-sdk-for-python/tree/main/sdk",
+    classifiers=[
+        "Development Status :: 4 - Beta",
```

Comment:
In keeping with "should we regenerate setup.py all the time", we could update development status to GA if we find the version is a GA version, or do you guys think it's a little out of the scope of this PR @lmazuel @msyyc

Reply:
https://github.com/Azure/autorest.python/pull/1154#discussion_r810401719